apps/web/vibes/soul/sections/product-detail/product-detail-form.tsx

```diff
@@ -166,10 +169,16 @@ function SubmitButton({ children, disabled }: { children: React.ReactNode; disab
 function FormField({
   field,
   formField,
+  prefetch,
 }: {
   field: Field;
   formField: FieldMetadata<string | number | boolean | Date | undefined>;
+  prefetch: boolean;
 }) {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
```

Comment:
We still seem to be using `useSearchParams` . Can't we use the result of `useQueryState` for this instead of `useSearchParams`?

I guess we're trying to get _all_ search params here, not just the ones being managed by `nuqs`?

Reply:
Yes, if I'm not wrong, I believe nuqs only manages the state param of this specific option, no?

From my testing, I couldn't get all params from this state we have in the field, unless there's another utility I'm missing?

Ultimately we want to get all params and just change the one for this field to the new option.

Reply:
Yeah, `nuqs` will only retrieve and change the params that have parsers defined. It does this so it doesn't clobber other search params that could have been set by something else, like pagination params or UTM params. It's a very nice design but you have to be aware of it. For example, in the PLP I had to handle pagination params in a special way so that changing PLP filters cleared pagination params. This meant I had to have `nuqs` "track" them so that they could be cleared when I called `setParams(null)`